Another run at cleaning up TODOs in refactored LLCP

erbr-ot opened the pull request:
Removing another bunch of TODOs 

This 'sits' on top of [PR#44550 ](https://github.com/zephyrproject-rtos/zephyr/pull/44550)

erbr-ot committed:
Bluetooth: test: adding helper to verify CTE_RSP notification

This to remove TODO and improve the CTE unit test

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot committed:
Bluetooth: controller: DO TODO, compile out members of struct re. CPR

Compile out misc. members not used when Connection Parameter Request
is not supported.
Implement missing tests re. unsupported features in CU/CPR procedure

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot committed:
Bluetooth: controller: removing unused functionality

Removing llcp_tx_flush() as it turns out it is not needed.

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot committed:
Bluetooth: controller: using REJECT_IND when EXT_IND was not negotiated

Unless a feature exchange determines support for REJECT_EXT_IND use
REJECT_IND for collision rejection

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot committed:
Bluetooth: controller: check for instant on reception of CONN_UPD_IND

Instead of waiting one connection interval, as a result of the state
change, perform the check for instant already on reception of
CONNECTION_UPDATE_IND

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

kruithofa commented on subsys/bluetooth/controller/ll_sw/ull_llcp_common.c:
```diff
@@ -621,6 +641,9 @@ static void lp_comm_rx_decode(struct ll_conn *conn, struct proc_ctx *ctx, struct
 		llcp_pdu_decode_reject_ext_ind(ctx, pdu);
 		break;
+	case PDU_DATA_LLCTRL_TYPE_REJECT_IND:
+		/* Empty on purpose, as we don't care about the PDU content, we'll disconnect */
```
why should we disconnect on receiving a REJECT_IND?

erbr-ot replied:
Argh, this should be part of PR44550, but a REJECT_IND here is not warranted here,  only REJECT_EXT in case we're running CTE procedure.

erbr-ot replied:
Ahh, sorry I'm confusing myself - I have the PR44550 commit here as well. 
So - forget me leading comment and consider the trailing one only;-)

erbr-ot commented:
rebased

thoh-ot commented on subsys/bluetooth/controller/ll_sw/ull_llcp.c:
```diff
@@ -795,27 +795,31 @@ uint8_t ull_cp_data_length_update(struct ll_conn *conn, uint16_t max_tx_octets,
 	ctx = llcp_rr_peek(conn);
-	if (ctx && (ctx->proc == PROC_ENCRYPTION_START || ctx->proc == PROC_ENCRYPTION_PAUSE)) {
+	if (ctx && (ctx->proc == PROC_ENCRYPTION_START || ctx->proc == PROC_ENCRYPTION_PAUSE) &&
+		llcp_rp_enc_ltk_req_reply_allowed(conn, ctx)) {
```
Minor: Indentation is wrong

erbr-ot committed:
Bluetooth: controller: implementing state check for ENC reply API

Now checking that encryption procedure is in correct state to accept
a reply from host.

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot commented:
fixed indent and rebased
